torch/distributed/fsdp/optim_utils.py

```diff
@@ -132,10 +132,10 @@ def _communicate_optim_state(
                 unpadded_numel = flat_param._orig_size.numel()  # type: ignore[attr-defined]
                 tensor_state[state_name] = tensor_buffer[:unpadded_numel].cpu()
         # Zero-dimension tensor state and non-tensor state: take this rank's
-        # value directly (`deepcopy()`ing to avoid aliasing surprises)
+        # value directly
         elif to_save:
             if _is_zero_dim_tensor(value):
-                zero_dim_tensor_state[state_name] = value
+                zero_dim_tensor_state[state_name] = value.cpu()
```

Comment:
Typically, this zero-dimension tensor is on CPU, but we should add a `.cpu()` call to ensure this.